### PR TITLE
Set accurate CMake project version and allow SameMajorVersion downstream

### PR DESCRIPTION
**Pull request by streeve**

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
0.6.0 does not seem right to me.
It does not convey that it is a development version, not the actual release.

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,8 +14,7 @@
 #------------------------------------------------------------------------------#
 cmake_minimum_required(VERSION 3.16)
 
-project(Cabana LANGUAGES CXX)
-set(PROJECT_VERSION "1.0-dev")
+project(Cabana LANGUAGES CXX VERSION 0.6.0)
```
Comment (dalg24) [inline CMakeLists.txt:17]:
```suggestion
project(Cabana LANGUAGES CXX VERSION 0.7.0-dev)
```

Comment (streeve) [reply CMakeLists.txt]: Makes sense, but apparently `project` doesn't allow non-numeric versions. I applied your suggestion to the previous version command

**Review by dalg24 — APPROVED** [review]

**Comment by streeve** [comment]
retest this please.

**Comment by streeve** [comment]
retest this please.

**Comment by streeve** [comment]
CI timeouts unrelated

**Pull request merged**